.github/workflows/build-release-assets.yml

```diff
@@ -6,35 +6,92 @@ on:
   workflow_dispatch:
   release:
     types: [published]
+
+env:
+  TEST_RESULTS: /tmp/test-results
+
 jobs:
-  releases-matrix:
-    name: Release Zip
+  build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        make_directive: [zip-x86_64, zip-arm64]
+    container:
+      image: cimg/go:1.19
     steps:
       - uses: actions/checkout@v3
-
-      - name: set up Go
-        uses: actions/setup-go@v3
+      - name: Set up test results directory
+        run: mkdir -p $GITHUB_WORKSPACE/test-results
+      - name: Run unit tests
+        run: |
+          PACKAGE_NAMES=$(go list ./... | xargs)
+          gotestsum --junitfile $TEST_RESULTS/gotestsum-report.xml -- $PACKAGE_NAMES
+      - name: Cache Go modules
+        uses: actions/cache@v3
         with:
-          go-version: 1.21
+          path: /go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Run build
+        run: make build
 
-      - name: make zip
-        run: make ${{ matrix.make_directive }}
-      
-      - name: upload artifact
-        env:
-          GH_TOKEN: ${{ secrets.RELEASE_ASSET_TOKEN }}
-        run: |
-          tag=$(gh release list --json tagName -q '.[0].tagName')
-          file=/tmp/newrelic-lambda-extension.arm64.zip
-          if [ -f /tmp/newrelic-lambda-extension.x86_64.zip ];then
-            file=/tmp/newrelic-lambda-extension.x86_64.zip
-          fi
+  coverage:
```

Comment:
can you move this and the unit tests to their own workflow file? These two things will always be run together.